doc/data-structures.rst

```diff
@@ -356,6 +353,14 @@ setting) variables and attributes:
 This is particularly useful in an exploratory context, because you can
 tab-complete these variable names with tools like IPython.
 
+.. warning::
+
+  We are changing the behavior of iterating over a Dataset (including
+  ``ds.keys()``) the next major release of xarray, to only include data
+  variables instead of both data variables and coordinates. In the meantime,
+  please prefer iterating over keys in ``ds.data_vars`` or ``ds.coords``, or
+  ``ds.variables`` if you need both at once.
```

Comment:
This could use some polishing. The first sentence should probably be split in two. Also, the "please prefer" statement is a bit awkward.

Reply:
I tried to keep this as a simple version of the full deprecation warning, and leave the detailed rationale to `whats-new.rst`. Let me know if this is any better...